intro-to-object-oriented-programming/defining-classes.md

```diff
@@ -0,0 +1,495 @@
+
+Many classes like to create objects with instances that have a customized, specific initial logic. In this situation, we may define a **special method** named `__init__()`, that will live inside the class definition.
+
+```python
+class ClassName:
+
+    def __init__(self):
+        pass
+```
+
+The `__init__` method:
+
+- Uses the exact same syntax for defining a function
+- Must be named `__init__`; Python looks for this exact name
+- Runs whenever we instantiate ("use") this class
+- Contains any logic that should be run whenever we instantiate ("use") this class
+- **Must always take at least one parameter,** because it's special
+  - The first parameter **always** represents an instance itself. Details below.
+  - This parameter is typically named `self`
+- Does not need to return anything (and should not return anything)
+
+For example, consider this `Album` class, that will print a message whenever an `Album` is instantiated:
+
+```python
+class Album:
+
+    def __init__(self):
+        print("Hello! I print whenever a new Album is initialized")
+```
+
+<!-- Question 2 -->
+
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: code-snippet
+* language: python3.6
+* id: Fshbd5
+* title: Constructor method
+
+##### !question
+
+Replace `pass` and add in the `__init__` method for this `Driver` class. In the `__init__` method, print out the string `"This init method runs every time a new Driver is instantiated!"`
+
+##### !end-question
+
+##### !placeholder
+
+```python
+class Driver:
+    pass
+```
+
+##### !end-placeholder
+
+##### !tests
+
+```py
+import unittest
+from main import *
+
+class TestDriverClass(unittest.TestCase):
+  def test_driver_is_defined(self):
+    self.assertTrue(Driver())
+```
+
+##### !end-tests
+
+#### !explanation
+
+Actually, this test will pass as long as there's a `Driver` class; it's challenging to write a test that verifies that the `__init__` method exists.
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+### !callout-info
+
+## "init" Is For Initialize
+
+We can describe the `__init__` method as a "class constructor," "constructor method," or "initializer." These names are used for any special method that runs once whenever an object is instantiated.
+
+### !end-callout
+
+## In Classes, We Can Access `self`
+
+In class definitions, in the `__init__` method and other instance methods, we will need to use a _special_ variable named `self`.
+
+Python wants to give instances the ability to reference itself with a variable. For example, as we write code to define the `Album` class, we may think about the "Dirty Computer" album. If we want to write code for "Dirty Computer" having its own state and behavior, we want a variable that represents "Dirty Computer" itself.
+
+This variable is almost always named `self`. Technically, this variable can have any name, but most Pythonists will use `self`.
```

Comment:
```suggestion
By convention, this parameter is almost always named `self`. Technically, the parameter could have any name, but most Pythonists stick with `self`.
```

Seeing how it looks without the last bit; I don't doubt they need motivation, but at this moment of curriculum, I want the conformity to be the virtue itself (not the empathizing with Pythonists sticking with something because it's what most other Pythonists expect).

Weirdly subtle, but I don't want a brain to go, "wait, should I be thinking about what other Pythonists expect? Am I not a Pythonist if I haven't been doing that?" (At this point in the curriculum, I'd want that questioning to stop at "readability is a virtue/principle in itself" and not a "Pythonists think this is readable so if you don't think it's readable then you're not doing something correctly")

Reply:
But there are plenty of readable names that could be used (this, me, myself). Even the original text says that most Pythonists use self, so it _is_ implying there's a convention. But I'm fine with not hitting the reader over the head with it. 😉